ql/src/java/org/apache/hadoop/hive/ql/exec/OperatorUtils.java

```diff
@@ -176,6 +176,37 @@ public static <T> Set<T> findOperatorsUpstreamJoinAccounted(Operator<?> start, C
     return found;
   }
 
+  /**
+   * Check whether there are more operators in the specified operator tree branch than the given limit
+   * @param start root of the operator tree to check
+   * @param opClazz type of operator to track
+   * @param limit maximum allowed number of operator in a branch of the tree
+   * @return true of limit is exceeded false otherwise
+   * @param <T> type of operator to track
+   */
+  public static <T> boolean hasMoreOperatorsThan(Operator<?> start, Class<T> opClazz, int limit) {
```

Comment:
Actually, I missed the fact that the method was counting operators in a certain **path** between nodes. I had the impression that just counting GBY operators reachable from a starting point was sufficient.
```
RS - GBY1 - GBY2 - FIL1 - \
                          JOIN
RS - GBY3 - FIL2 - - - - -/
```
Since we are calling the method on the inputs of the join (i.e., FIL1, FIL2) and not on the join itself I am not sure why we would need to backtrack and restart counting.

Anyways since we are mostly finalized the discussion on `hasMoreOperatorsThan` it may not worth jumping into implementing a new method.